ethereumj-core/src/main/java/org/ethereum/manager/BlockLoader.java

```diff
@@ -18,136 +18,196 @@
 
             if (block.getNumber() % 10 == 0) {
-                System.out.println(LocalDateTime.now().format(df) + " Imported block " + block.getShortDescr() + ": " + result + " (prework: "
-                        + exec1.getQueue().size() + ", work: " + exec2.getQueue().size() + ", blocks: " + exec1.getOrderMap().size() + ") in " +
-                        (System.currentTimeMillis() - s) + " ms");
+                LocalDateTime finish = LocalDateTime.now();
+
+                System.out.printf("%s Imported block %s: %s (prework: %d, work: %d, blocks: %d) in %d ms.\n",
+                        finish.format(df),
+                        block.getShortDescr(),
+                        result,
+                        exec1.getQueue().size(),
+                        exec2.getQueue().size(),
+                        exec1.getOrderMap().size(),
+                        System.currentTimeMillis() - start);
             }
 
-        } else {
-
-            if (block.getNumber() % 10000 == 0)
-                System.out.println("Skipping block #" + block.getNumber());
+        } else if (block.getNumber() % 10000 == 0) {
+            System.out.println("Skipping block #" + block.getNumber());
         }
     }
 
-    ExecutorPipeline<Block, Block> exec1;
-    ExecutorPipeline<Block, ?> exec2;
+    /**
+     * Tries import blocks from specified dumps.
+     *
+     * @param walkerFactory {@link DumpWalker} factory, which should instantiate new walker per each dump;
+     * @param paths         list of dumps to import;
+     * @return <code>true</code> if all blocks within all dumps have been successfully imported, <code>false</code> otherwise.
+     */
+    public boolean loadBlocks(Function<Path, DumpWalker> walkerFactory, Path... paths) {
+        Assert.notEmpty(paths, "There is nothing to import.");
 
-    public void loadBlocks() {
-        exec1 = new ExecutorPipeline(8, 1000, true, (Function<Block, Block>) b -> {
-            if (b.getNumber() >= blockchain.getBlockStore().getBestBlock().getNumber()) {
-                for (Transaction tx : b.getTransactionsList()) {
-                    tx.getSender();
-                }
-            }
-            return b;
-            }, throwable -> logger.error("Unhandled exception: ", throwable)
-        );
-
-        exec2 = exec1.add(1, 1000, block -> {
-            try {
-                blockWork(block);
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
-        });
+        initPipelines();
 
-        String fileSrc = config.blocksLoader();
+        AtomicLong maxBlockNumber = new AtomicLong();
+        boolean allBlocksImported;
         try {
-            final String blocksFormat = config.getConfig().hasPath("blocks.format") ? config.getConfig().getString("blocks.format") : null;
-            System.out.println("Loading blocks: " + fileSrc + ", format: " + blocksFormat);
-
-            if ("rlp".equalsIgnoreCase(blocksFormat)) {     // rlp encoded bytes
```

Comment:
We were working with 2 different formats of block dumps: 
1) Our own format, dump is fired by `record.blocks` parameter in config
2) Hive format, `rlp`

Maybe I'm missing something but I see only one possible format parsing in new code

Reply:
Yep, there were 2 dump formats. RLP is hive specific format, so I removed its support from ethereumj. Now ethereumj supports only own dump format. But if you need another one, you could implement DumpWalker interface and pass it to loadBlocks method for specific dump format reading.

Reply:
Ahhh I got it, cool!